mitodl/ocw-studio
# Page content UI

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #52 
Fixes #56 


#### What's this PR do?
Adds front-end pages for adding and editing site content. 

#### How should this be manually tested?
 - Listing content: go to `/sites/site-goes-here/` and click on Page or Resource in the sidebar. You should see a list of content with titles and an edit button on the side.
 - Add new content: click on "Add Page" or "Add Resource" on the upper right. Write in some simple content. If you're adding a page you should see the markdown editor. For resource you should just see a description textbox.
   - Leave title blank and click "Save". You should see a error message about a missing title.
   - Write in a title and click "Save". You should be returned to the content listing and you should see your new content at the bottom of the list. (If you refresh this will show up at the top of the list since the sort is by most recently modified).
 - Click "Edit" next to the item you just created. A drawer should open from the right and whatever you added before should appear there. 
   - You should be able to dismiss the drawer by clicking outside the drawer or clicking on the 'x' on the upper right.
   - Clear the title text field and click "Save". You should see a warning that the title is missing.
   - Make a change to the title and to the other content, and click "Save". The drawer should close and you should see that the content you modified has a new title in the list. If you click "Edit" next to it again the drawer should reopen and all of your changes that you saved should be there.

#### Other context
 - I left out mobile support since there are no wireframes yet
 - I'm not sure how permissions factor into the UI here so I left that out too

#### Screenshots
![Screenshot from 2021-03-03 08-51-35](https://user-images.githubusercontent.com/863262/109815589-cf4ffc00-7bfd-11eb-80e8-b4689d39cc19.png)

![Screenshot from 2021-03-02 16-44-29](https://user-images.githubusercontent.com/863262/109719398-9751a600-7b76-11eb-99c2-7802c23b5b5f.png)

![Screenshot from 2021-03-03 08-50-52](https://user-images.githubusercontent.com/863262/109815400-97e14f80-7bfd-11eb-8ce8-d538fe480209.png)
